src/main/scala/com/microsoft/hyperspace/util/LogicalPlanUtils.scala

```diff
@@ -35,4 +42,35 @@ object LogicalPlanUtils {
       case _ => false
     }
   }
+
+  def getAllAvailableIndexes(states: Seq[String] = Seq()): Seq[IndexLogEntry] = {
+    Hyperspace
+      .getContext(SparkSession.getActiveSession.get)
+      .indexCollectionManager
+      .getIndexes(states)
+  }
+
+  def getCandidateIndexesForPlan(plan: LogicalPlan): Seq[IndexLogEntry] = {
```

Comment:
Can we update this to
```scala
/**
 * Can we add a brief description of what this function does?
 */
def getCandidateIndexes(indexManager: IndexManager, plan: LogicalPlan): Seq[IndexLogEntry]
```
? The caller will still have to get active spark session, etc., which is fine.

Reply:
I also feel that this function should belong to rules (at least for now). Can we create `RuleUtils.scala` under `com.microsoft.hyperspace.index.rules`?

And let's also add a test suite for that.

Reply:
Ok, but it seems that common test code refactoring is required for Rules tests. (according to the comment in FilterIndexRuleTest 😊)
So how about filing a new issue for the refactoring and let this change be tested with the current tests?

Reply:
Rule tests refactoring is different from this. Also, I think this is a good ramp-up work for you anyway.

Reply:
I added RuleUtilsTest and revised the tests under rules a bit. Thanks!